network/src/peer_store/peer_store_impl.rs

```diff
@@ -165,20 +170,33 @@ impl PeerStore {
         let now_ms = ckb_systemtime::unix_time_as_millis();
         let peers = &self.connected_peers;
         let addr_expired_ms = now_ms.saturating_sub(ADDR_TRY_TIMEOUT_MS);
+
+        // Any protocol expect websocket
+        #[cfg(not(target_family = "wasm"))]
+        let filter = |peer_addr: &AddrInfo| {
+            extract_peer_id(&peer_addr.addr)
+                .map(|peer_id| !peers.contains_key(&peer_id))
+                .unwrap_or_default()
+                && peer_addr
+                    .connected(|t| t > addr_expired_ms && t <= now_ms.saturating_sub(DIAL_INTERVAL))
+                && required_flags_filter(required_flags, Flags::from_bits_truncate(peer_addr.flags))
+                && !matches!(find_type(&peer_addr.addr), TransportType::Ws)
+        };
+
+        // Only websocket on browser
+        #[cfg(target_family = "wasm")]
+        let filter = |peer_addr: &AddrInfo| {
+            extract_peer_id(&peer_addr.addr)
+                .map(|peer_id| !peers.contains_key(&peer_id))
+                .unwrap_or_default()
+                && peer_addr
+                    .connected(|t| t > addr_expired_ms && t <= now_ms.saturating_sub(DIAL_INTERVAL))
+                && required_flags_filter(required_flags, Flags::from_bits_truncate(peer_addr.flags))
+                && matches!(find_type(&peer_addr.addr), TransportType::Ws)
+        };
+
```

Comment:
suggest to move this code block to the fn `addr_manager#fetch_random`

Reply:
These filter conditions are not the same:
1. Get the feeler detection address list, because it ensures that the ws/tcp protocol addresses in the peer store can be updated normally
2. Get the identify attempt to connect to the address list formally, native only needs tcp, wasm only needs ws
3. Get the discovery propagation address list to ensure that any protocol is propagated

Reply:
the only diff is 
```
#[cfg(not(target_family = "wasm"))]
&& !matches!(find_type(&peer_addr.addr), TransportType::Ws
#[cfg(target_family = "wasm")]
&& matches!(find_type(&peer_addr.addr), TransportType::Ws)
```

other filter condition are same, we may move it to fetch_random, like:

```
  filter.filter(
#[cfg(not(target_family = "wasm"))]
&& !matches!(find_type(&peer_addr.addr), TransportType::Ws
#[cfg(target_family = "wasm")]
&& matches!(find_type(&peer_addr.addr), TransportType::Ws)
)
```

Reply:
`fetch_addrs_to_attempt` on native need `!matches!(find_type(&peer_addr.addr), TransportType::Ws)`, on WASM need `matches!(find_type(&peer_addr.addr), TransportType::Ws)`

The other two can be combined, but this function should not work.